osu.Server.Spectator.Tests/MultiplayerFlowTests.cs

```diff
@@ -475,6 +478,66 @@ public async void NotReadyUsersDontGetLoadRequest()
             Assert.Single(room.Users.Where(u => u.State == MultiplayerUserState.Idle));
         }
 
+        [Fact]
+        public async void UserCantChangeBeatmapAvailabilityWhilePlaying()
+        {
+            var room = await hub.JoinRoom(room_id);
+            var user = room.Users.Single();
+
+            await hub.ChangeState(MultiplayerUserState.Ready);
+            await hub.StartMatch();
+
+            // ensure throws when changing beatmap availability while loading.
+            await Assert.ThrowsAsync<InvalidStateException>(() => hub.ChangeBeatmapAvailability(new BeatmapAvailability(DownloadState.NotDownloaded)));
+
+            await hub.ChangeState(MultiplayerUserState.Loaded);
+
+            // ensure throws when changing beatmap availability while playing.
+            await Assert.ThrowsAsync<InvalidStateException>(() => hub.ChangeBeatmapAvailability(new BeatmapAvailability(DownloadState.NotDownloaded)));
+
+            await hub.ChangeState(MultiplayerUserState.FinishedPlay);
+
+            // ensure can change beatmap availability while in results, but state should be forced back to idle.
+            await hub.ChangeBeatmapAvailability(new BeatmapAvailability(DownloadState.NotDownloaded));
```

Comment:
this seems wrong? what if the user was still at the results screen? why should they be forced back to idle? they are now showing an incorrect state.

Reply:
Same as https://github.com/ppy/osu-server-spectator/pull/29#discussion_r554730608